functional_tests/steps/statistical_article_page.py

```diff
@@ -89,3 +118,62 @@ def expand_footnotes(context: Context):
 
     page.get_by_role("link", name="Footnotes").click()
     expect(footnotes_content).to_be_visible()
+
+
+@step("the user adds a correction")
+def user_adds_a_correction(context: Context):
+    page = context.page
+    page.wait_for_timeout(500)
+    page.locator("#tab-label-corrections_and_notices").click()
+    page.locator("#panel-child-corrections_and_notices-corrections-content").get_by_role(
+        "button", name="Insert a block"
+    ).click()
+    page.get_by_label("When*").fill("2025-03-13 13:59")
+    page.locator('[data-contentpath="text"] [role="textbox"]').fill("Correction text")
+    page.wait_for_timeout(500)
+
+
+@step("the user adds a notice")
+def user_adds_a_notice(context: Context):
+    page = context.page
+    page.wait_for_timeout(500)
+    page.locator("#tab-label-corrections_and_notices").click()
+    page.locator("#panel-child-corrections_and_notices-notices-content").get_by_role(
+        "button", name="Insert a block"
+    ).click()
+    page.get_by_label("When*").fill("2025-03-13 13:59")
+    page.locator('[data-contentpath="text"] [role="textbox"]').fill("Notice text")
+    page.wait_for_timeout(500)
+
+
+@then("the published statistical article page has the added correction")
+def the_published_statistical_article_page_has_the_added_correction(context: Context):
+    expect(context.page.get_by_role("link", name="Corrections ons-icon-chevron")).to_be_visible()
+    expect(context.page.get_by_text("13 March 2025 1:59p.m.")).to_be_hidden()
+    expect(context.page.get_by_text("Correction text")).to_be_hidden()
+
+
+@then("the published statistical article page has the added notice")
+def the_published_statistical_article_page_has_the_added_notice(context: Context):
+    expect(context.page.get_by_role("link", name="Notices ons-icon-chevron")).to_be_visible()
+    expect(context.page.get_by_text("13 March 2025 1:59p.m.")).to_be_hidden()
+    expect(context.page.get_by_text("Notice text")).to_be_hidden()
+
+
+@then("the user can edit the correction")
+def user_cannot_edit_the_correction(context: Context):
+    page = context.page
+    page.wait_for_timeout(500)  # added to allow JS to be ready
+    page.locator("#tab-label-corrections_and_notices").click()
+    expect(page.locator("#corrections-0-value-when")).to_be_editable()
+    page.wait_for_timeout(50)  # added to prevent flakiness
```

Comment:
Question: Why is a wait needed _after_ an assertion?

Reply:
I know this looks strange, but there is some weird race condition here where the test *following* this test would fail occasionally if there was no wait here. I don't want to do cargo cult programming here but this seems to have fixed it.

Reply:
Did you find that only occurs in this scenario? I wonder if we'd benefit from putting a very short wait like this in the after scenario hook, if this might potentially happen in other scenarios. Something to look at in a separate ticket.

Reply:
Thanks Adam, I've added some extra context to the comment now.